domain/permission/service/service.go

```diff
@@ -0,0 +1,185 @@
+		return errors.Trace(err)
+	}
+	return errors.Trace(s.st.DeletePermission(ctx, subject, target))
+}
+
+// ReadUserAccessForTarget returns the user access for the given user on
+// the given target. An error is returned if the given target's access
+// type does not exist.
+func (s *Service) ReadUserAccessForTarget(ctx context.Context, subject names.UserTag, target names.Tag) (permission.UserAccess, error) {
+	if err := validateTarget(target); err != nil {
+		return permission.UserAccess{}, errors.Trace(err)
+	}
+	userAccess, err := s.st.ReadUserAccessForTarget(ctx, subject, target)
+	return userAccess, errors.Trace(err)
+}
+
+// ReadUserAccessLevelForTarget returns the user access level for the
+// given user on the given target. An error is returned if the given
+// target's access type does not exist.
+func (s *Service) ReadUserAccessLevelForTarget(ctx context.Context, subject names.UserTag, target names.Tag) (permission.Access, error) {
+	if err := validateTarget(target); err != nil {
+		return "", errors.Trace(err)
+	}
+	access, err := s.st.ReadUserAccessLevelForTarget(ctx, subject, target)
+	return access, errors.Trace(err)
+}
+
+// ReadAllUserAccessForTarget return a slice of user access for all users
+// with access to the given target. An error is returned if the given
+// target's access type does not exist.
+func (s *Service) ReadAllUserAccessForTarget(ctx context.Context, target names.Tag) ([]permission.UserAccess, error) {
+	if err := validateTarget(target); err != nil {
+		return nil, errors.Trace(err)
+	}
+	userAccess, err := s.st.ReadAllUserAccessForTarget(ctx, target)
+	return userAccess, errors.Trace(err)
+}
+
+// ReadAllUserAccessForUser returns a slice of the user access the given
+// user has for any access type.
+func (s *Service) ReadAllUserAccessForUser(ctx context.Context, subject names.UserTag) ([]permission.UserAccess, error) {
+	userAccess, err := s.st.ReadAllUserAccessForUser(ctx, subject)
+	return userAccess, errors.Trace(err)
+}
+
+// ReadAllAccessTypeForUser return a slice of user access for the user
+// specified and of the given access type. An error is returned if the
+// given access type does not exist.
+// E.G. All clouds the user has access to.
+func (s *Service) ReadAllAccessTypeForUser(ctx context.Context, subject names.UserTag, accessType permission.AccessType) ([]permission.UserAccess, error) {
+	if err := accessType.Validate(); err != nil {
+		return nil, errors.Trace(err)
+	}
+	userAccess, err := s.st.ReadAllAccessTypeForUser(ctx, subject, accessType)
+	return userAccess, errors.Trace(err)
+}
+
+// UpdatePermission updates the user's access to the given target to the
+// given access.
+func (s *Service) UpdatePermission(ctx context.Context, subject names.UserTag, target names.Tag, access permission.Access) error {
+	var accessType permission.AccessType
+	// Unfortunately the tag kind values are not an exact
+	// match for permission access type values. Convert here.
+	switch target.Kind() {
```

Comment:
This conversion seems like something that should be moved to core/permission and done by the API Facade.

Reply:
done